testdata/p4_16_samples_outputs/issue2314-frontend.p4

```diff
@@ -0,0 +1,85 @@
+#include <core.p4>
+#define V1MODEL_VERSION 20180101
+#include <v1model.p4>
+
+header ethernet_t {
+    bit<48> dstAddr;
+    bit<48> srcAddr;
+    bit<16> etherType;
+}
+
+header H {
+    bit<8> a;
+}
+
+header I {
+    bit<16> etherType;
+}
+
+struct h {
+    ethernet_t ether;
+    H          h;
+    I          i;
+}
+
+struct m {
+}
+
+parser MyParser(packet_in b, out h hdr, inout m meta, inout standard_metadata_t std) {
+    bit<16> l3_etherType;
+    state start {
+        b.extract<ethernet_t>(hdr.ether);
+        transition L3_start;
+    }
+    state L3_start {
+        l3_etherType = hdr.ether.etherType;
+        transition L3_start_0;
+    }
+    state L3_start_0 {
+        transition select(l3_etherType) {
+            16w0x800: L3_h0;
+            16w0x8100: L3_i;
+            default: start_1;
+        }
+    }
+    state L3_h0 {
+        b.extract<H>(hdr.h);
+        transition start_1;
+    }
+    state L3_i {
+        b.extract<I>(hdr.i);
+        transition L3_start;
```

Comment:
I might be missing something, but this file issue2314-frontend.p4 seems to be showing the same bug that exists without this change: state L3_i transitions to L3_start, which makes an assignment to l3_etherType that should only be done when the sub parser is first called.  Should this transition be to L3_start_0 instead, perhaps?